internal/status/cache.go

```diff
@@ -38,22 +38,50 @@ const (
 func NewCache() Cache {
 	return Cache{
 		proxyUpdates: make(map[types.NamespacedName]*ProxyUpdate),
+		entries:      make(map[string]map[types.NamespacedName]CacheEntry),
 	}
 }
 
+type CacheEntry interface {
+	AsStatusUpdate() k8s.StatusUpdate
+	ConditionFor(ConditionType) *contour_api_v1.DetailedCondition
+}
+
 // Cache holds status updates from the DAG back towards Kubernetes.
 // It holds a per-Kind cache, and is intended to be accessed with a
 // KindAccessor.
 type Cache struct {
 	proxyUpdates map[types.NamespacedName]*ProxyUpdate
+
+	entries map[string]map[types.NamespacedName]CacheEntry
+}
+
+func (c *Cache) Acquire(obj k8s.Object) CacheEntry {
+	kind := k8s.KindOf(obj)
```

Comment:
`k8s.KindOf` apparently has a hardcoded map of type to kind, and `ExtensionService` is missing from it (so it's always returning the empty string).

Reply:
https://github.com/projectcontour/contour/blob/main/internal/k8s/kind.go#L47-L48

Also update the test to check this.

Reply:
guess i'm either blind or was looking at something stale. 🤷‍♂️